FogCreek/Glitch-Community
# [WIP] Splashy coil: componentize team page elements

partially blocked on https://glitch.manuscript.com/f/cases/3328470/add-whitelistedDomain-to-public-fields-for-team-API-results

## Links
* Remix link: https://splashy-coil.glitch.me
* Manuscript link: https://glitch.manuscript.com/f/cases/3328406/

## GIF/Screenshots
![Screen Shot 2019-05-01 at 1 51 45 PM](https://user-images.githubusercontent.com/938722/57032470-79345b00-6c18-11e9-9cf7-0b81a92c34bd.png)
![Screen Shot 2019-05-01 at 1 52 01 PM](https://user-images.githubusercontent.com/938722/57032471-79345b00-6c18-11e9-8a44-f9bb8d46d63d.png)
![Screen Shot 2019-05-01 at 1 52 14 PM](https://user-images.githubusercontent.com/938722/57032472-79345b00-6c18-11e9-9909-2467e4e4975d.png)
![Screen Shot 2019-05-01 at 1 52 40 PM](https://user-images.githubusercontent.com/938722/57032473-79345b00-6c18-11e9-852c-0d0de13f3327.png)

## Changes:
- componentize TeamUsers
- componentize or inline contents of team-elements
- extract team profile into TeamProfile component
- inline team fields into TeamProfile
- replace `{...props}` with explicit props passing
- replace helper methods on Team & Team editor with free functions
- refactor Team page as function component
- inline user-result-item into add-team-user-pop

## How To Test
(this is kind of involved, and requires two users)
* create a new team
* click 'add'
* write '@glitch.com' in the search box
* have another glitch user visit your team page
* the other user should be able to join
* you should be able to add / remove the other user's admin privileges, or remove them from the team
